multiple-observables/StatusObserver.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+
+namespace multiple_observables
+{
+    public class StatusObserver
+    {
+        public StatusObserver(IEnumerable<IComponent> components)
+        {
+            var statusObservables = components.Select(x => x.GetStatusObservable()).ToList();
+
+            // CombineLatest is used to get an observable that orchestrates multiple other observables.
+            // It fires when any of the underlying observables fire (and the predicate is satisfied as well).
+
+            // DistinctUntilChanged filters out observed values that are equal to their predecessors.
+
+            Observable.CombineLatest(statusObservables,
+                (lastStates) => lastStates.Any(status => status == Status.Error))
+                .DistinctUntilChanged()
+                .Subscribe(hasState => StatusError(hasState));
+                
+            Observable.CombineLatest(statusObservables,
+                (lastStates) => lastStates.All(status => status == Status.Ready))
```

Comment:
I recognized that it wouldn't work like it is now when you want to switch on/blink/off lights.

Reply:
Interesting. Functionally I did not make any changes. How should we proceed with this issue?

Reply:
I will show you how I would do it when I'm back home. After four more beers yesterday, I realized that I shouldn't turn off the light in two different methods which will be called in unknown order. So maybe the light will go off, but should stay on.

Reply:
Classic. Let's shift the discussion to monday. We should talk in person.

Reply:
With the new approach, that will work (I'm quite sure) we will get the cool switch expression back in code 😁

Reply:
Wow that's impressive! You should consider working as a professional software developer.